src/Reconciler.js

```diff
@@ -66,8 +84,15 @@ const hostConfig: HostConfig<
     domElement: Element,
     type: string,
     props: Props,
+    rootContainerInstance,
   ): boolean {
-    DOMComponent.setInitialProps(domElement, props);
+    if (isSvg == null) isSvg = getRootSvgContext(rootContainerInstance);
```

Comment:
this feels like it should unnecessary...I wonder if react-reconciler could pass the HostContext here? 

We need the context here because unlike react-dom, which has an "attributes first" approach, rdl is "property first" which also means we need to intentionally skip that path for SVGs, hence need to know if we are in an SVG context or not.

@gaearon I took a look at the reconciler code but it wasn't clear (to me) whether passing HostContext is possible at this stage of the commit. Do you have a sense of that?

Reply:
I think we don’t have host context during the commit phase.

During reconciliation, we go through every host component so we can track it.

During commit, we only go through the fibers marked as having “effects”. It’s a much shorter linked list of things that actually need DOM/lifecycle updates.

So we can’t have the current host context value during the commit because we jump over many nodes.

Reply:
Wait, I misunderstood. `finalizeInitialChildren` is *not* in the commit phase.

Only things called from `CommitWork` are commit phase. Things called from `BeginWork` or `CompleteWork` are reconciliation phase.

So yes, I think you could pass host context there if you want to. (If the reconciler doesn’t matter you can send a PR)